docs/architecture/state.md

```diff
@@ -1,91 +1,111 @@
+In this diagram, multiple participants interact with a common, publicly accessible state. The figure illustrates how notes are created and consumed:
 
-![Architecture core concepts](../img/architecture/state/nullifier-db.png)
+1. **Independent Transactions Creating Notes (tx1 & tx2):**  
+   Two separate users (Acc1 and Acc2) execute transactions independently:
+   - `tx1` produces **note 1**
+   - `tx2` produces **note 2**
 
-To prove that a note has not been consumed previously, the operator needs to provide a Merkle path to its node, and then show that the value in that node is `0`. In our case nullifiers are $32$ bytes each, and thus, the height of the Sparse Merkle Tree needs to be $256$.
+   These transactions occur in parallel and do not rely on each other, allowing concurrent processing without contention.
 
-To add new nullifiers to the database, operators need to maintain the entire nullifier set. Otherwise, they would not be able to compute the new root of the tree.
+2. **Sequencing and Consuming Notes (tx3):**  
+   A Miden node executes `tx3` against the shared account, consuming **notes 1 & 2** and producing **notes 3 & 4**. This merges independent contributions into a unified state update.
 
-> **Note**
-> Nullifiers as constructed in Miden break linkability of privately stored notes and the information about the note's consumption. To know the [note's nullifier](notes.md#note-nullifier-to-ensure-private-consumption) one must know the note's data.
+3. **Further Independent Transactions (tx4 & tx5):**  
+   After the shared state is updated:
+   - `tx4` consumes **note 4**
+   - `tx5` consumes **note 5**
+   
+   Both users can now interact with notes generated by the public account, continuing the cycle of state evolution.
 
-In the future, when the network experiences a large number of transactions per second (TPS), there will be one tree per epoch (~3 months), and Miden nodes always store trees for at least two epochs. However, the roots of the old trees are still stored. If a user wants to consume a note that is more than $6$ months old, there must be a merkle path provided to the Miden Node for verification.
+### State bloat minimization
 
-## State bloat minimization
+Operators do not need to know the entire state to verify or produce new blocks. Rather than storing the full state data with operators, users keep their data locally, and the rollup stores only commitments to that data. While some contracts must remain publicly visible, this approach minimizes state bloat and preserves privacy, since operators and other users only see hashes of user data.
```

Comment:
```suggestion
Miden nodes do not need to know the entire state to verify or produce new blocks. Rather than storing the full state data with the nodes, users keep their data locally, and the rollup stores only commitments to that data. While some contracts must remain publicly visible, this approach minimizes state bloat and preserves privacy, since nodes and other users only see hashes of user data.
```